matrix/spiral_print.py

```diff
@@ -10,8 +10,8 @@
 
 def checkMatrix(a):
     # must be
-    if type(a) == list and len(a) > 0:
-        if type(a[0]) == list:
+    if type(a) == tuple and len(a) > 0:
+        if type(a[0]) == tuple:
```

Comment:
```suggestion
    if a and if isinstance(a, Iterable):
```
PEP8 explains that we should not directly compare types.